inference-engine/src/transformations/src/transformations/convert_opset4_to_opset3/convert_sequences_to_sequences_ie.cpp

```diff
@@ -0,0 +1,208 @@
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(gru_sequence_ngraph, "ConvertGRUSequenceToGRUSequenceIE");
+    this->register_matcher(m, callback);
+}
+
+ngraph::pass::ConvertRNNSequenceMatcher::ConvertRNNSequenceMatcher() {
+    auto rnn_sequence_ngraph = ngraph::pattern::wrap_type<ngraph::opset4::RNNSequence>();
+
+    ngraph::matcher_pass_callback callback = [](pattern::Matcher &m) {
+        auto rnn_sequence = std::dynamic_pointer_cast<ngraph::opset4::RNNSequence>(m.get_match_root());
+        if (!rnn_sequence) {
+            return false;
+        }
+
+        auto W = std::dynamic_pointer_cast<ngraph::opset4::Constant>(
+                rnn_sequence->input_value(3).get_node_shared_ptr());
+        if (!W) {
+            return false;
+        }
+
+        auto R = std::dynamic_pointer_cast<ngraph::opset4::Constant>(
+                rnn_sequence->input_value(4).get_node_shared_ptr());
+        if (!R) {
+            return false;
+        }
+
+        // for forward/reverse cases we can squeeze num_direction dimension
+        auto axis_1 = ngraph::opset4::Constant::create(ngraph::element::i64, ngraph::Shape{1}, {1});
+        auto in_1 = std::make_shared<ngraph::opset4::Squeeze>(rnn_sequence->input(1).get_source_output(), axis_1);
+        auto concat = std::make_shared<ngraph::opset4::Concat>(ngraph::NodeVector({W, R}), 2);
+        auto axis_2 = ngraph::opset4::Constant::create(ngraph::element::i64, ngraph::Shape{1}, {0});
+        auto in_3 = std::make_shared<ngraph::opset4::Squeeze>(concat->output(0), axis_2);
+        auto in_4 = std::make_shared<ngraph::opset4::Squeeze>(rnn_sequence->input(5).get_source_output(), axis_2);
+        auto rnn_sequence_ie = std::make_shared<ngraph::op::RNNSequenceIE>(
+                rnn_sequence->input(0).get_source_output(),  // X
+                in_1,  // initial_hidden_state
+//                rnn_sequence->input(2).get_source_output(),  // sequence_lengths
+                in_3,                          // WR
+                in_4,  // B
+                rnn_sequence->get_hidden_size(),
+                rnn_sequence->get_direction(),
+                rnn_sequence->get_activations(),
+                rnn_sequence->get_activations_alpha(),
+                rnn_sequence->get_activations_beta(),
+                rnn_sequence->get_clip());
+
+        auto unsqueeze_axis = ngraph::opset4::Constant::create(ngraph::element::i64, ngraph::Shape{1}, {1});
+        auto unsqueeze_1 = std::make_shared<ngraph::opset4::Unsqueeze>(rnn_sequence_ie->output(0), unsqueeze_axis);
+        auto unsqueeze_2 = std::make_shared<ngraph::opset4::Unsqueeze>(rnn_sequence_ie->output(1), unsqueeze_axis);
+        NodeVector unsqueezes = {unsqueeze_1, unsqueeze_2};
+
+        for (int out_id = 0; out_id < rnn_sequence->outputs().size(); ++out_id) {
+            for (const auto& target_input : rnn_sequence->output(out_id).get_target_inputs()) {
+                target_input.replace_source_output(unsqueezes[out_id]);
```

Comment:
the same comment as above

Reply:
replaced with single replace_node() call